### PR TITLE
Add `win_wua.installed` to the win_wua module

### LINKED ISSUE #56640: Add ability to only list Updates that are installed.
Add ability to list only the updates that are installed on a System.
Also add the ability to scan in offline mode.

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds `win_wua.installed` function to win_wua module. The function will return a list of updates that apply to the current build of Windows.

### What issues does this PR fix or reference?
Fixes: #56640

### Previous Behavior
No way to get a list of installed updates

### New Behavior
Now you can get a list of installed updates

### Merge requirements satisfied?
- [x] Docs
- [x] Changelog
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by xeacott — DISMISSED** [review]

**Comment by Ch3LL** [comment]
Can we get an associated issue with this PR? Also there are some failing doc and pre-commit tests

File: salt/modules/win_wua.py [hunk 1]
```diff
@@ -169,39 +170,48 @@ def available(
             * Critical
             * Important
 
+        online (bool):
+            Tells the Windows Update Agent go online to update its local update
+            database. ``True`` will go online. ``False`` will use the local
+            update database as is. Default is ``False``
```
Comment (Ch3LL) [inline salt/modules/win_wua.py:176]: You state the default is False, but the kwarg is `online=True,`

File: salt/modules/win_wua.py [hunk 2]
```diff
@@ -270,35 +280,44 @@ def get(name, download=False, install=False):
             first to see if the update exists, then set ``install=True`` to
             install the update.
 
+        online (bool):
+            Tells the Windows Update Agent go online to update its local update
+            database. ``True`` will go online. ``False`` will use the local
+            update database as is. Default is ``False``
```
Comment (Ch3LL) [inline salt/modules/win_wua.py:286]: here as well. Default and kwarg conflict.

File: salt/modules/win_wua.py [hunk 3]
```diff
@@ -410,39 +431,48 @@ def list(
             * Critical
             * Important
 
+        online (bool):
+            Tells the Windows Update Agent go online to update its local update
+            database. ``True`` will go online. ``False`` will use the local
+            update database as is. Default is ``False``
```
Comment (Ch3LL) [inline salt/modules/win_wua.py:437]: here as well

File: salt/utils/win_update.py [hunk 4]
```diff
@@ -285,12 +287,21 @@ class WindowsUpdateAgent(object):
...
         Initialize the session and load all updates into the ``_updates``
         collection. This collection is used by the other class functions instead
         of querying Windows update (expensive).
 
+        Args:
+
+            online (bool):
+                Tells the Windows Update Agent go online to update its local
+                update database. ``True`` will go online. ``False`` will use the
+                local update database as is. Default is ``False``
```
Comment (Ch3LL) [inline salt/utils/win_update.py:301]: here

File: salt/utils/win_update.py [hunk 5]
```diff
@@ -333,12 +348,21 @@ def updates(self):
...
         Refresh the contents of the ``_updates`` collection. This gets all
         updates in the Windows Update system and loads them into the collection.
         This is the part that is slow.
 
+        Args:
+
+            online (bool):
+                Tells the Windows Update Agent go online to update its local
+                update database. ``True`` will go online. ``False`` will use the
+                local update database as is. Default is ``False``
```
Comment (Ch3LL) [inline salt/utils/win_update.py:362]: here

**Review by Ch3LL — CHANGES_REQUESTED** [review]
It looks like online's documentation and the kwarg conflict. I'm not sure what is suppose to be the default but I was wondering if it will change teh default behavior for people using this module?

File: salt/modules/win_wua.py (hunk 1 above)
Comment (twangboy) [reply salt/modules/win_wua.py]: Good catch!

**Comment by twangboy** [comment]
@Ch3LL There is no corresponding issue. It's a requirement for the VWin team.

**Comment by Ch3LL** [comment]
can we add one for tracking purposes? If someone wants to request the same feature, they will be able to see a feature request was already added.

Also just a clarification question. it looks like you made the default for online to be True. Does this change default behavior for users?

**Comment by twangboy** [comment]
Online True was the default. This adds the ability to scan without going out to Microsoft or a WSUS server.

**Comment by twangboy** [comment]
Linked to Issue

**Comment by xeacott** [comment]
This also links here ...
https://github.com/saltstack/release-planning/issues/197

**Comment by twangboy** [comment]
@xeacott That's actually for the `pending_reboot` grain

**Comment by xeacott** [comment]
Oh yeah, sorry put that in the wrong PR

**Comment by Ch3LL** [comment]
thanks for adding the issue @twangboy :)  

looks like there are some legit test failures here.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**